### ISSUE TITLE
How can I change the client name?

### ISSUE THREAD

**Issue by themihai** [anacrolix/torrent#191]
It seems some websites maintain a list with specific clients allowed so I'm wondering where is the client name specified and how can I change it.

**Comment by themihai** [comment]
I've found [this](https://github.com/anacrolix/torrent/blob/389898bb84b81c4e16d4dd1a7da3447246fe9788/tracker/http.go#L76) may be one place where the client name (User-Agent) is advertised (currently the default Go UA) . Is there any other place?

**Comment by anacrolix** [comment]
For the torrent client, https://github.com/anacrolix/torrent/blob/master/global.go#L21 . The convention is to prefix the peer ID with a string like that.

**Comment by themihai** [comment]
Would you consider a PR to define it in the Client Config? I believe the app developers should define  custom client IDs as each one is using the library differently. Another issue is that some trackers maintain whitelists with various clientIDs which makes the use of this lib impossible without changing the client ID(the reason why I opened this issue).

**Comment by anacrolix** [comment]
Yes, I would accept it as a Config option. Are you sure people whitelist clients? It's very easy to spoof, and seems silly that the solution is to just masquerade as another client.

**Comment by anacrolix** [comment]
You'd probably also want to make the UserAgent for anything HTTP related (like trackers) an option.

**Comment by themihai** [comment]
Yeah, I just stumbled over a such tracker(with clientID whitelists).  Just googled it and [found](http://www.torrent-invites.com/bittorrent-discussion/222786-private-trackers-torrent-client-allowed-banned-list-5.html) that the practice is widespread.  It seems to be poor man's method to ban selfish/abusing peers.  I will send a PR next week (unless someone else steps up).

**Comment by anacrolix** [comment]
Cheers, you're best positioned to implement it as you know what you require.
